File: LeetCode Solutions/Arrays 101/Valid Mountain Array/Valid Mountain Array.py
class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        if (len(A) < 3):
            return False
        i = 1
        while ((i < len(A)) and (A[i-1] < A[i])):
            i += 1
        if ((i == len(A)) or (i == 1)):
            return False
        while ((i < len(A)) and (A[i-1] > A[i])):
            i += 1
        return len(A) == i

File: LeetCode Solutions/Arrays 101/Valid Mountain Array/Valid Mountain Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& A) {
        if (A.size() < 3){
            return false;
        }
        int i = 1;
        while ((i < A.size()) && (A[i-1] < A[i])){
            i++;
        }
        if ((i == A.size()) || (i == 1)){
            return false;
        }
        while ((i < A.size()) && (A[i-1] > A[i])){
            i++;
        }
        return A.size() == i;
    }
};